supervisor/process.py

```diff
@@ -209,6 +211,9 @@ def spawn(self):
         self._assertInState(ProcessStates.EXITED, ProcessStates.FATAL,
                             ProcessStates.BACKOFF, ProcessStates.STOPPED)
 
+        if delaystart:
+            time.sleep(self.config.startdelay)
```

Comment:
> I need each program to get started N seconds after the previous one was started.

Another shell script solution would be something like this:

    #!/bin/sh
    supervisorctl start program1
    sleep 5
    supervisorctl start program2
    sleep 5
    supervisorctl start program3

You would then use this script to start them rather than `supervisorctl start all`.

It is also possible to add your own commands to `supervisorctl`.  You could add a custom start command.  It's fine to block inside `supervisorctl`, you just can't block inside `supervisord`.  I have an [example project](https://github.com/mnaberez/supervisor_cache) that shows how to add a command to `supervisorctl`, it's called a `ctlplugin`.